### ISSUE TITLE
google custom search content validation fails on Nu Html Checker 

### ISSUE THREAD

**Issue by markadlington** [validator/validator#467]
Hi 
I have google custom search on my web pages, and when I run Nu Html Checker validation on any of my web pages it fails with : two lines.. (This is the only failure on the web page, evrything else is clean)

> **Warning:** Element name gcse:search cannot be represented as XML 1.0.
> From line 81, column 1; to line 81, column 13
> `</script>↩<gcse:search></gcse>`

> **Error:** Element gcse:search not allowed as child of element div in this context. (Suppressing further errors from this subtree.)
> From line 81, column 1; to line 81, column 13
> `</script>↩<gcse:search></gcse`
> Content model for element div:

here is the actual code that is added to the web page 
(from `<script>` to `<gcse:search></gcse:search>` code is supplied by google.)

```html
<div id="searchform" >
<script>
  (function() {
    var cx = '014098160850134592381:-fwc_3sdlie';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>
<gcse:search></gcse:search>
</div>
```
Any ideas how to solve the problem?

**Comment by sideshowbarker** [comment]
See https://developers.google.com/custom-search/docs/element#html5

You can instead use the following:

```html
<div class="gcse-searchbox"></div>
```

**Comment by markadlington** [comment]
Unfortunatly the suggstion of using <div class="gcse-searchbox"></div> this breaks the google custome search (gcs) and 'gcs'  does not work
I have tried these two solutions below and both fail to show the google custom search in web page
any other ideas ??

**solution 1** 
<div id="searchform" >
<script>
  (function() {
    var cx = '014098160850134592381:-fwc_3sdlie';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>
<div class="gcse-searchbox"></div>
</div>

and
**solution 2** 
<div class="gcse-searchbox" id="searchform" >
<script>
  (function() {
    var cx = '014098160850134592381:-fwc_3sdlie';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>
<gcse:search></gcse:search>
</div>

**Comment by sideshowbarker** [comment]
> this breaks the google custom search (gcs) and 'gcs' does not work

Sorry then, I don’t know how to help you. What I posted is what Google’s own docs recommend.

Anyway, you can safely ignore the “Element gcse:search not allowed as child of element div in this context” error—or use the **Message filtering** button in the https://validator.w3.org/nu/ UI to persistently suppress it.

**Comment by bazzadp** [comment]
Is your problem with the GCS form? Or the results page?

For the form, you don't need to use the Google provided form (in fact I prefer not to so I get more control over the styling) so any standard HTML form will do, if you make the "action" load your search results page (which then picks up the search term from the URL parameters and displays your results).

You do lose some functionality (e.g. auto-complete) but nothing I've missed too much versus the gains of being in complete control of the form.

See here for example: https://www.tunetheweb.com/searchresults/?q=test

**Comment by sideshowbarker** [comment]
Closing per https://github.com/validator/validator/issues/467#issuecomment-279379277 but feel free to comment further here or re-open if you have a further resolution to propose

**Comment by LucasLarson** [comment]
You could force your document to validate by swapping out the following code from either of the [two proposals you provided](https://github.com/validator/validator/issues/467#issuecomment-279373861). The last three lines of them are both
```
</script>
<gcse:search></gcse:search>
</div>
```
but if you instead made them
```
document.write("<gcse:search><\/gcse:search>");
</script>
</div>
```
then your document will pass the validator.

**Comment by bazzadp** [comment]
https://stackoverflow.com/questions/802854/why-is-document-write-considered-a-bad-practice

**Comment by LucasLarson** [comment]
I definitely consider `document.write`‑ing, what should be included organically, harmful. It was just a solution for @markadlington, who I understood found document validation as paramount.
